File: docs/sphinx/snippets/python/examples/hello_bell_state.py
import cudaq

# [Begin Bell State Python]
@cudaq.kernel()
def bell(num_iters : int) -> int:
    q = cudaq.qvector(2)
    nCorrect = 0
    for i in range(num_iters):
        h(q[0])
        x.ctrl(q[0], q[1])
        results = mz(q)
        if results[0] == results[1]:
           nCorrect = nCorrect + 1
        
        reset(q) # reset the whole qvector
    return nCorrect

counts = bell(num_iters=100) # Pass as keyword arg for clarity
print(f'N Correct = {counts}')
assert counts == 100
# [End Bell State Python]

if __name__ == "__main__":
    pass # Logic is at top level

File: docs/sphinx/snippets/cpp/examples/hello_bell_state.cpp
#include <cudaq.h>
#include <stdio.h> // For printf

// [Begin Bell State C++]
struct bell {
  int operator()(int num_iters) __qpu__ {
    cudaq::qarray<2> q;
    int nCorrect = 0;
    for (int i = 0; i < num_iters; i++) {
      h(q[0]);
      x<cudaq::ctrl>(q[0], q[1]);
      auto results = mz(q);
      if (results[0] == results[1])
        nCorrect++;

      reset(q[0]);
      reset(q[1]);
    }
    return nCorrect;
  }
};

int main() {
  printf("N Correct = %d\n", bell{}(100));
  return 0;
}
// [End Bell State C++]
